import-scripts/study_directory_uses_updated_filenames.py

```diff
@@ -0,0 +1,538 @@
+    """
+    cancer_study_identifier = None
+    for filename in filenames:
+        if filename.startswith("meta_"):
+            filepath = os.path.join(basedir, filename)
+            if os.path.isfile(filepath):
+                cancer_study_identifier = determine_cancer_study_identifier_from_meta_file(filepath)
+        if cancer_study_identifier is not None:
+            # first success is adequate
+            return cancer_study_identifier
+    return None
+
+def split_wildcard_patterns(wildcard_patterns):
+    """
+    breaks a list of wildcard_patterns (each containing a single asterisk) into two parts, returning a list of dictionaries with prefix/suffix.
+    """
+    split_wildcard_pattern_list = []
+    for wildcard_pattern in wildcard_patterns:
+        pattern_segments = wildcard_pattern.strip().split("*")
+        if len(pattern_segments) != 2:
+            sys.stderr.write("Error : wildcard filename pattern '" + wildcard_pattern + "' does not have exactly one asterisk\n")
+            sys.exit(1)
+        split_wildcard_pattern = {"prefix" : pattern_segments[0], "suffix" : pattern_segments[1]}
+        split_wildcard_pattern_list.append(split_wildcard_pattern)
+    return split_wildcard_pattern_list
+
+def matches_wildcard_pattern(filename, split_wildcard_pattern):
+    """
+    returns True if filename starts with the prefix from, and ends with the suffix from, the split_wildcard_pattern dictionary.
+    """
+    return filename.startswith(split_wildcard_pattern["prefix"]) and filename.endswith(split_wildcard_pattern["suffix"])
+
+def matches_any_wildcard_pattern(filename, split_wildcard_patterns):
+    """
+    iterates through all pattern dictionaries, looking for any match for filename.
+    """
+    for split_wildcard_pattern in split_wildcard_patterns:
+        if (matches_wildcard_pattern(filename, split_wildcard_pattern)):
+            return True
+    return False
+
+def study_directory_uses_updated_filenames(path, cancer_study_identifier=None):
+    """
+    determines if unsupported filename patterns (those added June 2021) are present in a study directory
+
+    search directory at <path> - return true if any filenames are present, which:
+        - are not of the original set of filenames we recognized before the renaming effort, and
+        - would be imported according to the current datatypes sheet, and
+        - are not "normal sample" files (which are ignored by the importer)
+
+    arguments:
+    path -- the (absolute or relative) filename path to the directory which should be examined
+    cancer_study_identifier -- cancer study stable id [optional]. If not provided, the value is searched for in meta files within <path>
+    """
+    new_filenames_present = False
+    if os.path.isdir(path):
+        path = path + os.path.sep
+    basedir = os.path.dirname(path)
+    filenames = os.listdir(path)
+    if cancer_study_identifier is None:
```

Comment:
I think pythonic equivalent would be 
`if not cancer_study_idenfifier`

Reply:
done